src/ucp/core/ucp_worker.inl

```diff
@@ -44,6 +44,24 @@ static UCS_F_ALWAYS_INLINE ucs_status_t ucp_worker_rkey_config_get(
     return ucp_worker_add_rkey_config(worker, key, lanes_distance, cfg_index_p);
 }
 
+static UCS_F_ALWAYS_INLINE khint_t
+ucp_worker_mpool_hash_func(ucp_worker_mpool_key_t mpool_key)
+{
+    return (khint_t)mpool_key.mem_type ^ (mpool_key.sys_dev << 16);
```

Comment:
minor: `(khint_t)mpool_key.mem_type ^ (mpool_key.sys_dev << 8);`
since sysdev is small